src/app/static/src/app/components/header/LanguageMenu.vue

```diff
@@ -1,22 +1,14 @@
 <template>
-    <drop-down :text="currentLanguage" :right="true" style="flex: none">
-        <a class="dropdown-item" href="#" v-on:mousedown="() => changeLanguage('en')">
-            EN
-        </a>
-        <a class="dropdown-item" href="#" v-on:mousedown="() => changeLanguage('fr')">
-            FR
-        </a>
-        <a class="dropdown-item" href="#" v-on:mousedown="() => changeLanguage('pt')">
-            PT
-        </a>
-    </drop-down>
+    <language-menu-view :change-language="changeLanguage"
```

Comment:
Couple of small edits - is this inner component now needed? I'd revert this change to keep the component as it was originally.

And below when computing the current language, although this will work, we should technically use the `DataExplorationState`:

```
currentLanguage: mapStateProp<DataExplorationState, string>(null,
                (state: DataExplorationState) => state.language.toUpperCase())
```